steps_file.js

```diff
@@ -23,22 +24,29 @@ module.exports = function () {
 
     async emptyCart() {
       this.click(cartIcon);
-      const numOfElements = await this.grabNumberOfVisibleElements('//li[@class="product"]');
+      const numOfElements = await this.grabNumberOfVisibleElements(clearCartButton);
 
       if (numOfElements > 0) {
-        for (let i = 1; i < numOfElements * 2; i++) {
-          this.click('(//i[@class="linearicons-trash"])[last()]');
+        for (let i = 0; i < numOfElements; i++) {
+          this.click(clearCartButton);
         }
       }
     },
 
-    proceedToCheckout() {
-      this.click(addToCartButton);
-      this.click(cartIcon);
-    },
+    async clickCheckoutButton() {
+      try {
+        this.click(checkoutButton);
+        const notAvailabilityIndicatorIsVisible = await this.checkElementExists(notAvailableProduct);
 
-    clickCheckoutButton() {
-      this.click(checkoutButton);
+        if (notAvailabilityIndicatorIsVisible) {
+          throw new Error('Sorry, the product is not available');
```

Comment:
why do you need to throw an error to catch it in same function?
`if (await this.checkElementExists(notAvailableProduct)){
console.log('Please, choose the available product');
return false;
}
return true;`
or if remove console.log
`return await this.checkElementExists(notAvailableProduct)`